math/smatrix/inc/Math/CholeskyDecomp.h

```diff
@@ -590,7 +631,8 @@ namespace CholeskyDecompHelpers {
       void operator()(M& dst, const F* src, unsigned N) const
       {
          // make working copy
-         F * l = new F[N * (N + 1) / 2];
+         auto l_ = std::make_unique<F[]>(N * (N + 1) / 2);
+         F* l = l_.get();
```

Comment:
```suggestion
         std::vector<F> l(N * (N + 1) / 2);
```
Also this is a prime usecase for `std::vector` (with `.data()` in the line below).

Reply:
You're right, and I did consider this. It does mean including another header file. If we do end up preferring std::vector over the std::unique_ptr solution everywhere, this change will be made.